modules/distribution-zones/src/main/java/org/apache/ignite/internal/distributionzones/DistributionZoneManager.java

```diff
@@ -159,6 +167,36 @@ public class DistributionZoneManager implements IgniteComponent {
      */
     private final Map<Integer, ZoneState> zonesState;
 
+    /** Data nodes modification mutex. */
+    private final Object dataNodesMutex = new Object();
+
+    /** The last topology version wich was observed by distribution zone manager. */
+    private long lastTopVer;
+
+    /**
+     * Contains data nodes and meta info for zones.
+     * Map (zone id -> data nodes).
+     */
+    private final Map<Integer, DataNodes> dataNodes = new ConcurrentHashMap<>();
```

Comment:
If this and maps below are accessed only in synchronised block, do we really need `ConcurrentHashMap` here?

Reply:
Yes. I think we can use HashMap and TreeMap. Fixed.